source/ceylon/time/iso8601/parseTimeZone.ceylon

```diff
@@ -1,200 +1,54 @@
-                       else TzHours( sign, hours * 10 + characterToInteger(input.character), 2 );
-            }
-            else if ( input == ':' ) {
-                return TzColon( sign, hours );
-            }
-            else {
-                return TzError( "Unexpected character! Got '``input.character``' but expected a digit [0..9]" ); 
-            }
-        }
-        case( is EOF ) {
-            return digits == 2 
-                   then TzFinal( sign, hours ) 
-                   else TzError( "Unexpected end of input! Expected at two digits for hours but got one." );
-        }
+shared TimeZone? parseTimeZone( String offset ) {
+    if(offset == "Z") {
+        return timeZone.utc;
     }
     
-}
-
-"Represents the minutes, the accepted pattern is two digit minutes"
-class TzMinutes( Integer sign, Integer hours, minutes, digits = 1 ) extends TzState() {
-    Integer digits;
-    Integer minutes;
-    shared actual TzState next( Input input ) {
-        switch( input )
-        case ( is Chunk ) {
-            if( input.character.digit ) {
-                return digits == 2 
-                        then TzError( "Unexpected character! Got '``input.character``' but expected end of input" )	
-                        else TzMinutes( sign, hours, minutes * 10 + characterToInteger(input.character), 2 );
-            }
-            else {
-                return digits == 2 
-                        then TzError( "Unexpected character! Got '``input.character``' but expected end of input" )
-                        else TzError( "Unexpected character! Got '``input.character``' but expected a digit [0..9]" ); 
-            }
-        }
-        case ( is EOF ) {
-            return digits == 2 
-                       then TzFinal( sign, hours, minutes ) 
-                       else TzError( "Unexpected end of input! Expected two digits for minutes but got one." );
-        }
-    }
-
-}
-
-"Represents the colon as some patterns accepts for example 'hh:mm'"
-class TzColon( Integer sign, Integer hours ) extends TzState() {
-    shared actual TzState next( Input input ) {
-        switch( input )
-        case ( is Chunk ) {
-            return input.character.digit 
-                   then TzMinutes( sign, hours, characterToInteger( input.character ) )
-                   else TzError( "Unexpected character! Got '``input.character``' but expected a digit [0..9] after ':'" ); 
-        }
-        case ( is EOF ) {
-            return TzError( "Unexpected end of input! Expecting a digit [0..9] after ':'" );
-        }
-    }
-}
-
-"Represents the parser successfully finished"
-class TzFinal( Integer sign = 1, Integer hours = 0, Integer minutes = 0 ) extends TzState() {
-    shared actual TzState next(Input character) => this;
-    shared Integer evaluate() {
-        return sign * ( hours * ms.perHour + minutes * ms.perMinute );
+    value signal =
+            switch(offset[0])
+    case ('+')  1
```

Comment:
Humm... it seems there is some issue with ceylon formatter with indenting this syntax?